systemd/systemd.go

```diff
@@ -652,19 +667,19 @@ func (c *Collector) newDbus() (*dbus.Conn, error) {
 	return dbus.New()
 }
 
-func filterUnits(units []dbus.UnitStatus, whitelistPattern, blacklistPattern *regexp.Regexp) []dbus.UnitStatus {
-	filtered := make([]dbus.UnitStatus, 0, len(units))
+func filterUnits(units []dbus.UnitStatus, whitelistPattern, blacklistPattern *regexp.Regexp) chan dbus.UnitStatus {
```

Comment:
Consider not using channel here, I would rather have `filterUnits` return a list of units as it is simpler to understand and makes more sense (i.e. it takes a list and filters it returning a filtered list)

I do believe that this method isn't a performance bottleneck, but the actual metric collection for unit is.

Reply:
Seems reasonable. At most I'm getting about 15ms of speedup by making `filterUnits` concurrent, which is not much. The initial `ListUnits` is where the bulk of the serial startup time occurs. If a concurrent API is ever made for systemd.dbus then it might make sense to revisit making our consumption concurrent, but the win is very small at the moment